### PR TITLE
Refactored LaTeX generation in preview

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed

Refactored the interactive LaTeX generation a bit so that the LaTeX preamble etc are defined in a single place.

#### Other comments

As an added bonus one can specify fontsize as an int, but I do not think that warrants a release note. Although it is mentioned in the documentation.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
I think this looks good although I'm not really sure how to test it. I don't think I've seen `preview` work on OSX.

**Comment by oscargus** [comment]
All the LaTeX generation (like in IPython and Spyder) goes through `preview`  and it is only the LaTeX-file generation that has been refactored (and slightly changed).

(I have problems testing the other aspects of preview as well.)

**Comment by asmeurer** [comment]
`preview()` works just fine on Mac if you configure it to save to a file. The only issue is with the default Pyglet viewer. IMO pyglet shouldn't be the default (for preview or plotting).

Ideally `viewer='open'` (meaning use the `open` command, which would open it in Preview.app) ought to work and be the default on Mac, but it doesn't for me (it opens Preview but no image is opened). 

See also https://github.com/sympy/sympy/pull/13621 and https://github.com/sympy/sympy/issues/15680

**Comment by oscargus** [comment]
I was thinking about adding a `'system'` viewer, but that requires platform-specific code and when I didn't get it to work on Windows (because the default viewers returns after starting causing the temporary file to be deleted before it is loaded in the viewer...), I didn't spend more time on it...

**Comment by asmeurer** [comment]
Doing NotImplementedError on Windows for now would be fine. I will say my experience on Windows in general is that you shouldn't delete files that are open in programs like you would in Unix, so you'd have to find a way to only delete the file after the program closes. Perhaps Windows has some API for this?

**Comment by oscargus** [comment]
Except from that the feature request from @asmeurer about an ability to open a file in the system viewer is not implemented, this PR does what it originally was supposed to do: generate the LaTeX-file in a single place and use the same LaTeX-file for different cases. I think it may just as well go into 1.11.

**Comment by oscarbenjamin** [comment]
Not sure what the circleci error is. I guess that's because this PR predates using circleci.

@asmeurer does the circleci build use the PR branch or does it use the branch resulting from merging the PR into master (like most other things in CI do)?

**Comment by oscarbenjamin** [comment]
@oscargus I haven't looked through this in detail but it looks fine. Feel free to merge if you're happy with it.

File: sympy/printing/preview.py [hunk 1]
```diff
@@ -78,12 +79,51 @@ def on_expose():
 
     win.close()
 
+def _get_latex_main(expr, *, preamble=None, packages=(), preamble_lines=(),
```
Comment (oscargus) [inline sympy/printing/preview.py:82]:
```suggestion
def _get_latex_main(expr, *, preamble=None, packages=(), preamble_lines=None,
```

**Comment by oscargus** [comment]
I changed the name of `preamble_lines` to `extra_preamble` to reduce the confusion if it is actually lines (as in a tuple or list) or just a string.

**Comment by oscargus** [comment]
I rebased and then the CircleCI stuff was OK (I guess this was pretty old).

**Comment by oscarbenjamin** [comment]
I think something funny happened with the rebase because it's showing extra commits that were already on master.

**Comment by oscargus** [comment]
Ahh! Sorry about that! And automerge on top of that...

**Pull request merged**